src/main/java/org/adoptopenjdk/jheappo/heapdump/StackFrame.java

```diff
@@ -32,13 +32,12 @@
     long sourceFileNameStringID;
     long classSerialNumber;
 
-    public StackFrame(byte[] body) {
-        super(body);
-        stackFrameID = extractID();
-        methodNameStringID = extractID();
-        methodSignatureStringID = extractID();
-        sourceFileNameStringID = extractID();
-        classSerialNumber = extractID();
+    public StackFrame(EncodedChunk body) {
```

Comment:
I'm wondering if this approach is fragile, relying on a sequence of the same method call to get the next result...

Reply:
It absolutely is. I'm not really sure of a way to address this other than tests for deserializing each data type.

Reply:
the whole lets read in a binary encoded file is "fragile". So much so that if you make it to the end of the file without some error you're pretty much assured you've done it correctly. It's an odd property of fragile

Reply:
Maybe this is a crazy idea as it does duplicate code by what if you had 'properly' named function calls with each of them delegating to `extractID();`

e.g. 

```
methodSignatureStringID = extractMethodSignatureID();
sourceFileNameStringID = extractSourceFileNameID();
classSerialNumber = extractClassSerialNumbert();

.....

extractSourceFileNameID() {
   extractId();
}


```

Reply:
Hmm... still depends on the order being right, does it not?

Perhaps we should move to having the caller pull out numbers and pass them to the constructor? Or having a pattern of factory static methods that make the strict order perhaps easier to visually see?

Reply:
I like the factory static methods idea, are you happy to give that a go?

Reply:
I can do that, but it'll induce less merge pain if we sort out the rest of the PRs one way or another and then do it separately